File: 1297-maximum-number-of-balloons/maximum-number-of-balloons.js
/**
 * @param {string} text
 * @return {number}
 */
var maxNumberOfBalloons = function(text) {
    function count(l){
        let c = 0
        for(let i of text){
            if(i==l){
                c+=1
            }
        }
        return c
    }
    let a = Math.min(count("a"),count("b"),count("n"))
    let b = Math.min(count("l"),count("o"))
    return Math.min(Math.floor(b/2),a)
};

File: 1297-maximum-number-of-balloons/maximum-number-of-balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        a = min([text.count("a"),text.count("b"),text.count("n")])
        if a == 0:
            return 0
        b = min([text.count("l"),text.count("o")])
        if b <= 1:
            return 0
        if b//2 >= a:
            return a
        return b//2
        
        